tls/s2n_x509_validator.c

```diff
@@ -371,6 +371,39 @@ s2n_cert_validation_code s2n_x509_validator_validate_cert_chain(struct s2n_x509_
 
         S2N_ERROR_IF(op_code <= 0, S2N_ERR_CERT_UNTRUSTED);
         validator->state = VALIDATED;
+
+        /* Check to ensure certificate signatures appear in certificate_signature_preferences.
+         * Additionally we only want this check to apply in TLS1.3 */
```

Comment:
This method is already quite long. I would move this chunk of validation to a separate method.